Data/Primitive/Array.hs

```diff
@@ -34,6 +34,7 @@ import qualified GHC.Exts as Exts
 import GHC.Exts (fromListN, fromList)
 #endif
 
+import qualified Data.Foldable as F
```

Comment:
This import is only used within the scope of an `#if MIN_VERSION_base(4,9,0) ... #endif` block, so on old GHCs this is an unused import warning. I think moving the import next to `Data.Semigroup` (which is appropriately guarded by CPP) should fix this.